Trappist/src/Promact.Trappist.Repository/TestSettings/TestSettingsRepository.cs

```diff
@@ -1,42 +1,34 @@
-﻿using Promact.Trappist.DomainModel.DbContext;
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using Promact.Trappist.DomainModel.ApplicationClasses.TestSettings;
+using Promact.Trappist.DomainModel.DbContext;
 using Promact.Trappist.DomainModel.Models.Test;
+using System;
 using System.Linq;
-
+using System.Threading.Tasks;
 
 namespace Promact.Trappist.Repository.TestSettings
 {
     public class TestSettingsRepository : ITestSettingsRepository
     {
         private readonly TrappistDbContext _dbContext;
-        Test test = new Test();
+
 
         public TestSettingsRepository(TrappistDbContext dbContext)
         {
             _dbContext = dbContext;
         }
 
         /// <summary>
-        /// Updates the changes made to the Settings of a Test
+        /// Updates the changes made to the settings of a Test
         /// </summary>
-        /// <param name="id">The parameter "id" is used to access the Settings of that Test</param>
-        /// <param name="testObject">The parameter "testObject" is used as an object of the Model Test</param>
-        /// <returns>Updated Setting of that Test</returns>
-        public string UpdateTestSettings(int id, Test testObject)
+        /// <param name="testACObject">The parameter "testACObject" is an object of TestSettingsAC</param>
+        /// <returns>Updated Settings of that Test</returns>
+        public async Task<string> UpdateTestSettingsAsync(TestSettingsAC testACObject)
         {
-            var settingsToUpdate = _dbContext.Test.FirstOrDefault(check => check.Id == id);
-            settingsToUpdate.TestName = testObject.TestName;
-            settingsToUpdate.BrowserTolerance = testObject.BrowserTolerance;
-            settingsToUpdate.StartDate = testObject.StartDate;
-            settingsToUpdate.EndDate = testObject.EndDate;
-            settingsToUpdate.Duration = testObject.Duration;
-            settingsToUpdate.FromIpAddress = testObject.FromIpAddress;
-            settingsToUpdate.ToIpAddress = testObject.ToIpAddress;
-            settingsToUpdate.WarningMessage = testObject.WarningMessage;
-            settingsToUpdate.CorrectMarks = testObject.CorrectMarks;
-            settingsToUpdate.IncorrectMarks = testObject.IncorrectMarks;
-            settingsToUpdate.WarningTime = testObject.WarningTime;
-            _dbContext.Test.Update(settingsToUpdate);
-            _dbContext.SaveChanges();
+            Test settings = Mapper.Map<TestSettingsAC, Test>(testACObject);
+            _dbContext.Test.Update(settings);
+            await _dbContext.SaveChangesAsync();
             return "Data Updated";
```

Comment:
We never really return string to mention that data was updated. For methods where Update of record has taken place it usually does not return anything.